optimum/exporters/tasks.py

```diff
@@ -792,6 +804,54 @@ class TasksManager:
     _UNSUPPORTED_CLI_MODEL_TYPE = {"unet", "vae-encoder", "vae-decoder", "clip-text-model", "trocr"}
     _SUPPORTED_CLI_MODEL_TYPE = set(_SUPPORTED_MODEL_TYPE.keys()) - _UNSUPPORTED_CLI_MODEL_TYPE
 
+    @classmethod
+    def create_register(
+        cls, backend: str, overwrite_existing: bool = False
+    ) -> Callable[[str, Tuple[str, ...]], Callable[[Type], Type]]:
+        """
+        Creates a register function for the specified backend.
+
+        Args:
+            backend (`str`):
+                The name of the backend that the register function will handle.
+            overwrite_existing (`bool`, defaults to `False`):
+                Whether or not the register function is allowed to overwrite an already existing config.
+
+        Returns:
+            `Callable[[str, Tuple[str, ...]], Callable[[Type], Type]]`: A decorator taking the model type and a the
+            supported tasks.
+
+        Example:
+            ```python
+            >>> register_for_new_backend = create_register("new-backend")
+
+            >>> @register_for_new_backend("bert", "sequence-classification", "token-classification")
+            >>> class BertNewBackendConfig(NewBackendConfig):
+            >>>     pass
+            ```
+        """
+
+        def wrapper(model_type: str, *supported_tasks: str) -> Callable[[Type], Type]:
```

Comment:
Will the supported tasks differ a lot among different backend? It could be a long list if I register every task for every model in this manner.

Reply:
Maybe.
It will be like that:

```python
@register("bert", "sequence-classification", "token-classification", "multiple-choice", "question-answering")
class BertNeuronConfig(...):
...
```

But you can also define global variables and re-use them:
```python
COMMON_TEXT_TASKS = ("sequence-classification", "token-classification", "multiple-choice", "question-answering")

@register("bert", *COMMON_TEXT_TAKS)
class BertNeuronConfig(...):
...
```